

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>NGS Pipeline Guide &mdash; dolphin 1.0 documentation</title>
  

  
  

  

  
  
    

  

  
  
    <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  

  

  
    <link rel="top" title="dolphin 1.0 documentation" href="../index.html"/>
        <link rel="up" title="Dolphin-UI" href="Dolphin-UI.html"/>
        <link rel="next" title="NGS Status Guide" href="ngsstatus.html"/>
        <link rel="prev" title="Excel Export Guide" href="excelexport.html"/> 

  
  <script src="../_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="../index.html" class="icon icon-home"> dolphin
          

          
          </a>

          
            
            
              <div class="version">
                1.0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
                <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../dolphin-docker/Dolphin-Docker.html">Dolphin-Docker</a></li>
<li class="toctree-l1"><a class="reference internal" href="../dolphin-tools/Dolphin-Tools.html">Dolphin-Tools</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="Dolphin-UI.html">Dolphin-UI</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="excelimport.html">Excel Import Guide</a></li>
<li class="toctree-l2"><a class="reference internal" href="fastlane.html">Fastlane Guide</a></li>
<li class="toctree-l2"><a class="reference internal" href="ngsbrowser.html">NGS Browser Guide</a></li>
<li class="toctree-l2"><a class="reference internal" href="excelexport.html">Excel Export Guide</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="">NGS Pipeline Guide</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#getting-started">Getting Started</a></li>
<li class="toctree-l3"><a class="reference internal" href="#ngs-pipeline-parameters">NGS Pipeline Parameters</a></li>
<li class="toctree-l3"><a class="reference internal" href="#optional-parameters">Optional Parameters</a></li>
<li class="toctree-l3"><a class="reference internal" href="#additional-pipelines">Additional Pipelines</a></li>
<li class="toctree-l3"><a class="reference internal" href="#submission">Submission</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="ngsstatus.html">NGS Status Guide</a></li>
<li class="toctree-l2"><a class="reference internal" href="ngsreports.html">NGS Reports Guide</a></li>
<li class="toctree-l2"><a class="reference internal" href="ngsplots.html">NGS Plots Guide</a></li>
<li class="toctree-l2"><a class="reference internal" href="tablecreator.html">Table Creator Guide</a></li>
</ul>
</li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="../index.html">dolphin</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          





<div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="../index.html">Docs</a> &raquo;</li>
      
          <li><a href="Dolphin-UI.html">Dolphin-UI</a> &raquo;</li>
      
    <li>NGS Pipeline Guide</li>
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="../_sources/dolphin-ui/ngspipeline.txt" rel="nofollow"> View page source</a>
          
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="ngs-pipeline-guide">
<h1>NGS Pipeline Guide<a class="headerlink" href="#ngs-pipeline-guide" title="Permalink to this headline">¶</a></h1>
<p>This guide will walk you through all of your options within the Run Pipeline page.</p>
<div class="section" id="getting-started">
<h2>Getting Started<a class="headerlink" href="#getting-started" title="Permalink to this headline">¶</a></h2>
<p>First, make sure to have an instance of dolphin available (see Dolphin Docker) as well as an account for the dolphin interface.</p>
<p>Once logged in, click on the &#8216;NGS Tracking&#8217; tab on the left, then click on &#8216;NGS Browser&#8217;.</p>
<img alt="../_images/menu_bar.png" class="align-center" src="../_images/menu_bar.png" />
<p>Once you have selected the samples/imports you wish to analyze, hit the &#8216;Send to Pipeline&#8217; at the bottom of the screen to begin.</p>
<p>Alternatively, if you&#8217;re rerunning a run with different parameters, you will be taken to the NGS Pipeline page.</p>
</div>
<div class="section" id="ngs-pipeline-parameters">
<h2>NGS Pipeline Parameters<a class="headerlink" href="#ngs-pipeline-parameters" title="Permalink to this headline">¶</a></h2>
<p>Once you&#8217;ve reached the pipeline page, there should be a table at the top of the page with your selection of samples from the browser.</p>
<p>You can explore this table further at your discretion, however the main portion of the page lies under the table within the tabs.</p>
<p>These tabs describe the parameters to the run you will be creating.</p>
<p>The first few tabs are required parameters, and they include:</p>
<p><strong>Name the Run:</strong></p>
<p>This parameter is a brief name you will be giving your run.</p>
<p>This will help you quickly identify which run is which within the status page.</p>
<p><strong>Description:</strong></p>
<p>This parameter is a brief description of the current run you will be creating.</p>
<p>This may help clear up any confusion that might be caused about the run in question later.</p>
<p><strong>Genome Build:</strong></p>
<p>From this dropdown menu, you&#8217;ll be able to select the specific genome build you wish to use.</p>
<p><strong>Mate-paired:</strong></p>
<p>Use the dropdown menu to select whether or not your libraries are mate paired or not.</p>
<p><strong>Resume Run:</strong></p>
<p>Select whether this is a fresh run, or a continuation of a previous run.</p>
<p><strong>Output Directory:</strong></p>
<p>This determines where the specific output of the current run should be deposited within the system.</p>
<p>If you&#8217;re not entirely sure as where to deposit your data, you can contact your local administrator.</p>
<p><strong>FastQC:</strong></p>
<p>Select whether or not you want a FastQC report.</p>
</div>
<div class="section" id="optional-parameters">
<h2>Optional Parameters<a class="headerlink" href="#optional-parameters" title="Permalink to this headline">¶</a></h2>
<p>Further down the page, there are tabs with a plus sign on them.</p>
<img alt="../_images/optional_param.png" class="align-center" src="../_images/optional_param.png" />
<p>By clicking on the plus sign, you expand the tab and have aditional parameter options to choose from.</p>
<img alt="../_images/optional_param_open.png" class="align-center" src="../_images/optional_param_open.png" />
<p>The expandable tabs include:</p>
<p><strong>Adapter Removal:</strong></p>
<p>If your samples require adapter removal, you can expand the Adapter Removal tab and select the yes button.</p>
<p>Once you&#8217;ve selected yes, you can then add your specific adapters within the adapter text box.</p>
<p>There should be one adapter per line within the text box.</p>
<p><strong>Quality Filtering:</strong></p>
<p>If your samples require quality filtering, you can expand the Quality Filtering tab and select the yes button.</p>
<p>After you&#8217;ve selected yes, you then can adjust the following quality filtering parameters:</p>
<ul class="simple">
<li>window size</li>
<li>required quality</li>
<li>leading</li>
<li>trailing</li>
<li>minlen</li>
</ul>
<p><strong>Split FastQ:</strong></p>
<p>If you would like the to split the resulting fastq files, you can expand the Split FastQ and select the yes button.</p>
<p>Once yes is selected, you then can select the size of the fastq file based on the number of reads per file.</p>
<p><strong>Trimming:</strong></p>
<p>If you would like to trim the reads within your samples selected, you can do so by expanding the Trimming tab and selecting the yes button.</p>
<p>Next, select whether or not the reads are paired-end or single-end.</p>
<p>After you&#8217;ve selected your read type, then enter the 5&#8217; length to trim and the 3&#8217; length to trim.</p>
<p>If paired-end reads are selected, additionally you need will need to supply the 5&#8217; and the 3&#8217; length of the second read too be trimmed.</p>
<p><strong>Custom Sequence Set:</strong></p>
<p>If you would like to add custom sequence sets, expand the Custom Sequence Set and select the &#8216;Add a Custom Sequence Set&#8217; button.</p>
<p>For these custom sequence sets, you will have to supply:</p>
<ul class="simple">
<li>The custom sequence index file</li>
<li>Name of the index</li>
<li>Bowtie parameters</li>
<li>Description of the index</li>
<li>Whether to filter out the results</li>
</ul>
<p>The full path of the index file is required for custom sequence sets.</p>
<p>You can add multiple custom sequence sets if desired.</p>
<p><strong>Common RNAs:</strong></p>
<p>If you would like to map your reads sequentially to common RNAs, Expand the Common RNAs tab and select the yes for the RNAs you would like to map.</p>
<p>Bowtie2 map your reads sequentially to common RNAs below. To change the default parameters, click the <cite>Change Parameters</cite> button and then insert your parameters.</p>
</div>
<div class="section" id="additional-pipelines">
<h2>Additional Pipelines<a class="headerlink" href="#additional-pipelines" title="Permalink to this headline">¶</a></h2>
<p>If you would like to add additional features to your current run, you can expand the Additional Pipelines tab and hit the &#8216;Add Pipeline&#8217; button.</p>
<p>You can add more than one selection of additional pipelines by clicking the &#8216;Add Pipeline&#8217; button again.  You can also remove your current pipeline selection by clicking the &#8216;Remove Pipeline&#8217; button.</p>
<p>A new box will appear with a dropdown menu that includes:</p>
<p><strong>RNASeqRSEM:</strong></p>
<p>Selecting the RNASeqRSEM additional pipeline will run RSEM in addition to the current run.</p>
<p>RSEM is an additional pipeline that will estimate gene and isoform expression levels for RNA-Seq data.</p>
<p>Upon selection of RNASeqRSEM, you have the option of selecting additional RSEM command line parameters as well as IGC/TDF or BigWig conversions.</p>
<p><strong>Tophat:</strong></p>
<p>Selection the Tophat additional pipeline will allow you to run Tophat in addition to the current run.</p>
<p>Tophat is a popular RNA-seq alignment tool used for mapping and splice junction detection.</p>
<p>Upon selecting the Tophat pipeline addition, you have the option of adding additional tophat command line parameters as well as IGC/TDF or BigWig conversions.</p>
<p><strong>ChipSeq:</strong></p>
<p>If your samples of interest include ChipSeq data, then the ChipSeq additional pipeline should be selected.</p>
<p>After selection of ChipSeq, some additional parameters are required.</p>
<p>Chip Input Definitions are the names of the files of your chipseq input.  Multimappers determines the maximum number of locations reads are allowed to map to.</p>
<p>Tag size, in base pairs, for MACS determines the size of the tags while the Band Width, in base pairs, for MACS determines the size of the sequenced regions.</p>
<p>The Effective genome size, in base pairs, is the size of the mappable part of the genome.</p>
<p>IGV/TDF and BigWig conversion is also selectable.</p>
<p><strong>DESeq:</strong></p>
<p>First, in order to select the DESeq additional pipeline option, you must have already selected the RNASeqRSEM option first.</p>
<p>The DESeq pipeline allows for differential expression analysis to be conducted amongst your samples.</p>
<p>Using the selection boxes labeled &#8216;Condition 1&#8217; and &#8216;Condition 2&#8217; you can select which samples you wish to check against.</p>
<p>Once you&#8217;ve selected your conditions, you then can determine your Fit Type, and p-adjustment cutoff value as well as whether or not you want the Heatmap and the Fold Change cutoff.</p>
<p>Based on your previous selection from the Common RNAs tab, you can also select which given sequences you want to analyze.</p>
<p>Note that you can select DESeq multiple times, incase you want to run multiple pairwise comparisons on a single run.</p>
</div>
<div class="section" id="submission">
<h2>Submission<a class="headerlink" href="#submission" title="Permalink to this headline">¶</a></h2>
<p>Once all of your parameters are squared away and you&#8217;ve selected all of the additional options/pipelines that you desire for your run, you may hit the &#8216;Submit&#8217; button at the bottom left of the page to submit the run.</p>
</div>
</div>


           </div>
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="ngsstatus.html" class="btn btn-neutral float-right" title="NGS Status Guide" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="excelexport.html" class="btn btn-neutral" title="Excel Export Guide" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2015, Alper Kucukural, Nick Merowsky, Alistair Firth.

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'../',
            VERSION:'1.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="../_static/jquery.js"></script>
      <script type="text/javascript" src="../_static/underscore.js"></script>
      <script type="text/javascript" src="../_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="../_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>